pkg/cmd/admin/router/router.go

```diff
@@ -438,7 +440,8 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 			return fmt.Errorf("router could not be created; you must specify a service account with --service-account")
```

Comment:
All of the service account code should be moved into a method that ensures the service account first, verifies it exists, and checks it (it should not update if dry run is true).  This method is getting enormous.